File: vltdemo-java/src/main/java/com/verizon/location/platformdemo/MapTrafficDemo.java
package com.verizon.location.platformdemo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.Toast;

import com.verizon.location.commonmodels.Coordinate;
import com.verizon.location.maps.MapFeature;
import com.verizon.location.maps.MapInitializationError;
import com.verizon.location.maps.MapView;
import com.verizon.location.maps.OnMapReadyCallback;
import com.verizon.location.maps.VltMap;
import com.verizon.location.maps.VltMapOptions;

import org.jetbrains.annotations.NotNull;

public class MapTrafficDemo extends AppCompatActivity {

    private boolean mapIsVisible;
    private VltMapOptions options = new VltMapOptions();
    private VltMap map;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_traffic_demo);

        options.setZoom(12);
        options.setTarget(new Coordinate(39.7557f,-104.9942f));

        MapView mv = findViewById(R.id.map_view);
        mv.onCreate(savedInstanceState);
        mv.initialize(getResources().getString(R.string.map_key), options, new OnMapReadyCallback() {
            @Override
            public void onMapFailedToLoad(@NotNull MapInitializationError mapError) {
                Log.e("MAP ERROR", ""+mapError);
                Toast.makeText(MapTrafficDemo.this, "" + mapError.getErrorDescription(), Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onMapReady(@NotNull VltMap map) {
                MapTrafficDemo.this.map = map;
                mapIsVisible = true;
            }
        });

        CheckBox trafficFlowCheckBox = findViewById(R.id.feature_switch_traffic_flow);
        trafficFlowCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                if (mapIsVisible) {
                    map.setFeatureVisible(MapFeature.TRAFFIC_FLOW, b);
                    if (b) {
                        trafficFlowCheckBox.setText(getResources().getString(R.string.traffic_flow_on_label));
                    } else {
                        trafficFlowCheckBox.setText(getResources().getString(R.string.traffic_flow_off_label));
                    }
                }
            }
        });

        CheckBox trafficIncidentsCheckBox = findViewById(R.id.feature_switch_traffic_incidents);
        trafficIncidentsCheckBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                if (mapIsVisible) {
                    map.setFeatureVisible(MapFeature.TRAFFIC_INCIDENTS, b);
                    if (b) {
                        trafficIncidentsCheckBox.setText(getResources().getString(R.string.traffic_incident_on_label));
                    } else {
                        trafficIncidentsCheckBox.setText(getResources().getString(R.string.traffic_incident_off_label));
                    }
                }
            }
        });
    }

    @Override
    public boolean onSupportNavigateUp() {
        super.onBackPressed();
        this.finish();
        return true;
    }
}

File: vltdemo/src/main/java/com/verizon/location/platformdemo/MapTrafficDemo.kt
package com.verizon.location.platformdemo

import android.os.Bundle
import android.widget.CheckBox
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.verizon.location.commonmodels.Coordinate
import com.verizon.location.maps.*
import timber.log.Timber

class MapTrafficDemo : AppCompatActivity() {

    private var vltMap: VltMap? = null
    private var vltMapOptions = VltMapOptions()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_map_traffic)

        findViewById<CheckBox>(R.id.feature_switch_traffic_flow)?.let {
            it.setOnCheckedChangeListener { compoundButton, b ->
                vltMap?.apply {
                    if (it.isChecked) {
                        this.setFeatureVisible(MapFeature.TRAFFIC_FLOW, true)
                        it.text = resources.getString(R.string.traffic_flow_on_label)
                    } else {
                        this.setFeatureVisible(MapFeature.TRAFFIC_FLOW, false)
                        it.text = resources.getString(R.string.traffic_flow_off_label)
                    }
                }
            }
        }

        findViewById<CheckBox>(R.id.feature_switch_traffic_incidents)?.let {
            it.setOnCheckedChangeListener { compoundButton, b ->
                vltMap?.apply {
                    if (it.isChecked) {
                        this.setFeatureVisible(MapFeature.TRAFFIC_INCIDENTS, true)
                        it.text = resources.getString(R.string.traffic_incident_on_label)
                    } else {
                        this.setFeatureVisible(MapFeature.TRAFFIC_INCIDENTS, false)
                        it.text = resources.getString(R.string.traffic_incident_off_label)
                    }
                }
            }
        }

    val mv = findViewById<MapView>(R.id.map_view)
        mv.onCreate(savedInstanceState)

        vltMapOptions.target = Coordinate(42.3637f, -71.053604f)
        vltMapOptions.zoom = 12.0

        mv.initialize(
            resources.getString(R.string.map_key),
            vltMapOptions,
            object : OnMapReadyCallback {
                override fun onMapReady(map: VltMap) {
                    vltMap = map
                }

                override fun onMapFailedToLoad(mapError: MapInitializationError) {
                    Timber.e("map failed to load %s", mapError.errorDescription)
                    Toast.makeText(
                        this@MapTrafficDemo,
                        "map failed to load : $mapError.errorDescription",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            })
    }

    override fun onSupportNavigateUp(): Boolean {
        super.onBackPressed()
        finish()
        return true
    }

}